lib/jellygrinder/client/helpers/utils.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Jellygrinder.Client.Helpers.Utils do
+  @moduledoc false
+
+  alias Jellygrinder.Client.Helpers.ConnectionManager
+  alias Jellygrinder.Coordinator
+
+  @spec request(String.t(), String.t(), map()) :: {:ok | :error, binary()}
+  def request(path, label, state) do
+    timestamp = get_current_timestamp_ms()
+    start_time = System.monotonic_time()
+    maybe_response = ConnectionManager.get(state.conn_manager, path)
+    end_time = System.monotonic_time()
+
+    request_info = %{
+      timestamp: timestamp,
+      elapsed: System.convert_time_unit(end_time - start_time, :native, :millisecond),
+      label: label,
+      process_name: state.name,
+      path: path
+    }
+
+    {result, data} =
+      case maybe_response do
+        {:ok, response} ->
+          success = response.status == 200
+          data = Map.get(response, :data, "")
+
+          {%{
+             response_code: response.status,
+             success: success,
+             failure_msg: if(success, do: "", else: data),
```

Comment:
This success field feels redundant to me. We have both `response_code`, which indicates success, and the existence of `failure_msg` also indicates whether it is a success or not.

Reply:
It is a bit redundant, yes, but we use it as a boolean in a few places, and if we want to get rid of it, we'd have to compare response_code with 200 in each of them. But I suppose that could be done

Reply:
Maybe if we use it in a few places we should make a struct from it and make an API for getting information from this report?